dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MultipleBackJumpsToForwardJumps.java

```diff
@@ -0,0 +1,78 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.program.Function;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.core.CondJump;
+import com.dat3m.dartagnan.program.event.core.Label;
+
+import static java.util.Comparator.comparingInt;
+
+import java.util.List;
+import java.util.function.ToIntFunction;
+
+/*
+    This pass transforms loops to have a single backjump using forward jumping.
+    Given a loop of the form
+
+    L:
+    ...
+    if X goto L
+    ...
+    if Y goto L
+    More Code
+
+    it transforms it to
+
+    L:
+    ...
+    if X goto Forward_to_L
+    ...
+    if Y goto Forward_to_L
+    goto Continue_after_last
+    Forward_to_L:
+    goto L
+    Continue_after_last
+    More Code
+    ...
+*/
+public class MultipleBackJumpsToForwardJumps implements ProgramProcessor {
+
+    public static MultipleBackJumpsToForwardJumps newInstance() {
+        return new MultipleBackJumpsToForwardJumps();
+    }
+
+    @Override
+    public void run(Program program) {
+        program.getFunctions().forEach(f -> backToForwardJumps(f, Event::getGlobalId));
+    }
+
+    private void backToForwardJumps(Function function, ToIntFunction<Event> linId) {
+        for (Label label : function.getEvents(Label.class)) {
+            final List<CondJump> backJumps = label.getJumpSet().stream()
+                    .filter(j -> linId.applyAsInt(j) > linId.applyAsInt(label))
+                    .toList();
+
+            if (backJumps.size() > 1) {
+                final CondJump last = backJumps.stream().max(comparingInt(linId)).get();
```

Comment:
`LoopFormVerification` requires that the unique backjump is also unconditional and this pass may fail to achieve that if `backjumps.size()==1`. The check `backjumps.size() == 1 && backjumps.get(0).isGoto()` would be fine. 
Btw. the only language level construct that can produce conditional backjumps are `do-while` loops.

Reply:
You mean `backJumps.size() > 1 || (backjumps.size() == 1 && backjumps.get(0).isGoto())`, right?

I could also use `backJumps.size() > 0`. This will do the simplification even for the case where we have a single unconditional backjump, but I don't expect this to create any performance issue and it keeps the code simpler.

Reply:
I meant to skip the code if my proposed conditions holds:
```
if (backjumps.size() == 1 && backjumps.get(0).isGoto()) {
   return;
}
```
Technically there is no problem in always doing this transformation but it has the ugly consequence that the pass is not idempotent but I think it should be just for sanity reasons.